miloyip/rapidjson
# Fix racy install of cmake package config files

This is to fix an issue that we observed in an Yocto linux environment when building rapidjson. The installation included cmake package configuration files that pointed to the absolution paths in the original build environment.

Upon further investigation the issue turned out to be a racy installation of such files within rapidjson. This patch should fix the issue.

--

**Manual testing**

Before the change the install target would output:
```
-- Installing: /.../install/lib/cmake/RapidJSON/RapidJSONConfig.cmake
-- Up-to-date: /.../install/lib/cmake/RapidJSON/RapidJSONConfig.cmake
-- Installing: /.../install/lib/cmake/RapidJSON/RapidJSONConfigVersion.cmake
```
showing the attempt to install the same file twice.

After the change the install target would output only:
```
-- Installing: /.../install/lib/cmake/RapidJSON/RapidJSONConfig.cmake
-- Installing: /.../install/lib/cmake/RapidJSON/RapidJSONConfigVersion.cmake
```

--

The content of the installed file is as expected. References to include directory are relative to the package config file location:
```
# Compute paths
get_filename_component(RapidJSON_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)

set( RapidJSON_INCLUDE_DIR  "${RapidJSON_CMAKE_DIR}/../../../include" )
set( RapidJSON_INCLUDE_DIRS  "${RapidJSON_CMAKE_DIR}/../../../include" )
message(STATUS "RapidJSON found. Headers: ${RapidJSON_INCLUDE_DIRS}")
```
